### DISCUSSION TITLE
explicit-module-boundary-types with allowHigherOrderFunctions allowed formats

### DISCUSSION THREAD

**Discussion by bscaspar** [typescript-eslint/typescript-eslint#7061] [Technical Discussions]
### Body

Here's the relevant snippet of my `.eslintrc.json`:
```json
"rules": {
     "@typescript-eslint/explicit-function-return-type": [
      "warn",
      {
        "allowHigherOrderFunctions": true
      }
    ],
}
```
With this config I am encountering the following:

```js
// ❌ fails 
const arrowFn1 = (): void => {
  return;
};
export const arrowHoc = () => arrowFn1;

// ✅ passes
export const arrowFn2 = () => (): void => {
  return;
};
```
Is this expected? Thanks for any input!

### Additional Info

Any additional info...

### Before you submit your discussion, please confirm the following. If any of these required steps are not taken, we may not be able to review your RFC. Help us to help you!

- [X] I have [read the discussions guidelines](https://typescript-eslint.io/contributing/discussions) and per those guidelines, this fits a discussion category, _not_ a help request or standard issue.
- [X] I have [searched for related discussions](https://github.com/typescript-eslint/typescript-eslint/discussions) and [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues) and found none that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

**Comment by bradzacher** [comment]
As per our contributing guide https://typescript-eslint.io/contributing/discussions

We do not use discussions as a support forum. Please read the guide to find alternatives.
